### PR TITLE
Create Account and Stock services

### PR DESCRIPTION
**Pull request by boraelci**
- Created Account service
- Created Stock service
- Updated AccountServiceTest to use doReturn().when() format
- Changed 4 space tabs to 2 spaces in AccountServiceTest

### CODE REVIEW

File: src/main/java/com/ase/restservice/controller/AccountController.java [hunk 1]
```diff
@@ -42,33 +45,33 @@ public Account createAccount(@Valid @RequestBody Account account) {
...
    * @param accountId AccountID
    * @param amount Value that will be summed with balance
-   * @return Updated account.
+   * @return Updated account
    * @throws ResourceNotFoundException if account does not exist in the database
    */
   @PutMapping("/accounts/{accountId}/balance")
-  public Account updateAccountBalance(@PathVariable(value = "accountId")
-      String accountId, @RequestParam(value = "amount", defaultValue = "0") String amount)
+  public Account updateAccountBalance(@PathVariable(value = "accountId") String accountId,
```
Comment (jlamborn324) [inline src/main/java/com/ase/restservice/controller/AccountController.java:63]: The logic in this function seems like it should be in the service layer

File: src/main/java/com/ase/restservice/service/AccountService.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  @Autowired
+  private AccountRepository accountRepository;
+
+  /**
+   * Saves an account to the database.
+   *
+   * @param account Account
+   * @return Updated account
+   */
+  public Account save(Account account) {
```
Comment (jlamborn324) [inline src/main/java/com/ase/restservice/service/AccountService.java:24]: Could be a more descriptive name? like createNewAccount

File: src/main/java/com/ase/restservice/controller/AccountController.java (hunk 1 above)
Comment (boraelci) [reply src/main/java/com/ase/restservice/controller/AccountController.java:63]: Changed it

File: src/main/java/com/ase/restservice/service/AccountService.java (hunk 2 above)
Comment (boraelci) [reply src/main/java/com/ase/restservice/service/AccountService.java:24]: Changed this after implementing your other comment. Let's have separate create, update, delete methods for the services. I also created an interface to enforce this

**Pull request merged**